src/tools/store.cpp

```diff
@@ -87,8 +91,22 @@ bool generateDataStoreOptions(const int argc,
     return true;
 }
 
+[[ noreturn ]] void CleanupSharedBarriers(int signum)
+{ // Here the lock state of named mutexes is unknown, make a hard cleanup
+    osrm::storage::SharedBarriers::resetCurrentRegions();
+    osrm::storage::SharedBarriers::resetRegions1();
```

Comment:
right, this is dangerous, but reaching this point with a custom or default handler is a bad sign for the whole IPC because locks will be in unknown state. I will remove reseting regions, so it can be controlled by `--max-wait` and reset at https://github.com/Project-OSRM/osrm-backend/blob/master/src/storage/storage.cpp#L158-L164